File: ruby/src/list.rb
class ListNode
    attr_accessor :val, :next
    def initialize(val)
        @val = val
        @next = nil
    end
end

def create_list(max)
	head = ListNode.new 0
	(1..max).inject(head) { |head, val| head.next = ListNode.new(val); head = head.next }
	head
end

def print_list(head)
	until head.nil?
		print head.val, ' '
		head = head.next
	end

	puts
end

File: javascript/src/list.js
function ListNode(val) {
	this.val = val;
	this.next = null;
};

function createOddList(max) {
	var head = new ListNode(0);
	var node = head;

	for (var i = 1; i < max; ++i)
	{
		node.next = new ListNode(i * 2 + 1);
		node = node.next;
	}

	return head;
};

function createEvenList(max) {
	var head = new ListNode(0);
	var node = head;

	for (var i = 1; i < max; ++i)
	{
		node.next = new ListNode(i * 2);
		node = node.next;
	}

	return head;
};

function createList(max) {
	var head = new ListNode(1);
	var node = head;

	for (var i = 2; i < max; ++i)
	{
		node.next = new ListNode(i);
		node = node.next;
	}

	return head;
};

function printList(list) {
	var str = '';
	while (list != null)
	{
		str += list.val;
		str += ' ';
		list = list.next;
	}
	console.log(str);
};

exports.createOddList = createOddList;
exports.createEvenList = createEvenList;
exports.createList = createList;
exports.printList = printList;
